### PR TITLE
README.md: improve documentation and add examples

### PR DESCRIPTION
**Pull request by knusbaum**
This PR improves the README with better instructions and examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,38 +4,250 @@ Automatic instrumentation of Go code
...
+
+Orchestrion processes Go source code and automatically inserts instrumentation.
+Currently this instrumentation will create Datadog APM traces for the instrumented code, but future work will include support for OpenTelemetry as well.
+
 ## Getting started
 
+Orchestrion will automatically add instrumentation to your source code. This process saves the time that engineers would have to spend manually adding the tracer and integrations by replacing that process with a single command. There are a couple ways it's possible to add instrumentation to your projects using orchestrion.
+
+There are a couple ways to use orchestrion
+Orchestrion will also optionally (`-rm`) remove any instrumentation it has added, so this process is reversible.
```
Comment (drichards-87) [inline README.md:17]:
```suggestion
There are a couple of ways to add instrumentation to your projects using orchestrion. Orchestrion will also optionally (`-rm`) remove any instrumentation it has added, so this process is reversible.
```

File: README.md [hunk 2]
```diff
@@ -4,38 +4,250 @@ Automatic instrumentation of Go code
...
+
+### Instrument your code before check-in.
```
Comment (drichards-87) [inline README.md:19]:
```suggestion
### Instrument your code before check-in
```

**Comment by knusbaum** [comment]
Thanks, @drichards-87. I incorporated all of your suggestions.

**Comment by drichards-87** [comment]
Thanks @knusbaum. I'm going to let @brett0000FF take it over from here.

**Review by brett0000FF — APPROVED** [review]
Looks good! Just some minor edits for you to consider.

As we discussed offline, 'll make a note elsewhere to investigate adding this to our public docs when this goes GA.

**Review by drichards-87 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -4,38 +4,248 @@ Automatic instrumentation of Go code
...
 
 ```sh
+## Go to your project root
+cd ~/my_go_project
+
+## Run orchestrion on the root directory. This will recursively instrument everything in the directory and its subdirectories.
 orchestrion -w ./
+
+## Build your project like normal. If you get errors here, you may need to tidy your go.mod file (go mod tidy)
+go build ./cmd/exe
```
Comment (ahmed-mez) [inline README.md:39]: nit: the docs team might have some guidelines - I just think it's better to provide generic examples
```suggestion
go build ./path/to/entrypoint
```

**Review by ahmed-mez — APPROVED** [review]

**Pull request merged**